### PR TITLE
G-API: Removing G-API test code that is a reflection of ts module

### PR DESCRIPTION
**Pull request by mpashchenkov**
This PR sets the second parameter of FindDataFile to false for tests of G-API. This handles the problem with failed tests which aren't skipped.
Also removing `initTestDataPath()` logic.
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### Build Configuration
```
force_builders=Custom,Custom Win,Custom Mac
build_gapi_standalone:Linux x64=ade-0.1.1f
build_gapi_standalone:Win64=ade-0.1.1f
build_gapi_standalone:Mac=ade-0.1.1f
build_gapi_standalone:Linux x64 Debug=ade-0.1.1f

Xbuild_image:Custom=centos:7
Xbuildworker:Custom=linux-1
build_gapi_standalone:Custom=ade-0.1.1f

build_image:Custom=ubuntu-openvino-2021.4.1:20.04
build_image:Custom Win=openvino-2021.4.1
build_image:Custom Mac=openvino-2021.4.1

test_modules:Custom=gapi,python2,python3,java
test_modules:Custom Win=gapi,python2,python3,java
test_modules:Custom Mac=gapi,python2,python3,java

buildworker:Custom=linux-1
# disabled due high memory usage: test_opencl:Custom=ON
test_opencl:Custom=OFF
test_bigdata:Custom=1
test_filter:Custom=*
```

### CODE REVIEW

File: modules/gapi/test/cpu/gapi_ocv_stateful_kernel_tests.cpp
Comment (alalek) [inline modules/gapi/test/cpu/gapi_ocv_stateful_kernel_tests.cpp:195]: `false` should not be used for files from `opencv_extra`.
`false` is used for files from external additional **optional** share.

**Review by alexgiving — APPROVED** [review]

File: modules/gapi/test/cpu/gapi_ocv_stateful_kernel_tests.cpp
Comment (mpashchenkov) [reply modules/gapi/test/cpu/gapi_ocv_stateful_kernel_tests.cpp:183]: Can we add a check for `OPENCV_TEST_DATA_PATH` for emptiness? FAILED tests without `OPENCV_TEST_DATA_PATH` can shock people.

Comment (alalek) [reply modules/gapi/test/cpu/gapi_ocv_stateful_kernel_tests.cpp:183]: > FAILED tests without OPENCV_TEST_DATA_PATH can shock people

This is expected behavior.
Launching tests without required test data doesn't make sense. They should not pass at least.
https://github.com/opencv/opencv/wiki/QA_in_OpenCV

Comment (sivanov-work) [reply modules/gapi/test/cpu/gapi_ocv_stateful_kernel_tests.cpp:183]: @alalek  can you clarify this assertion? i faced with the same behavior in my PR and got several comments.
Could you take a look please?
https://github.com/opencv/opencv/pull/20901/files#diff-df9d13b0d9035428deb2704ecdd722cb4a3193ba653554215768156fffec1f62R43

TO my mind it is expected behavior for:
1) throw SkipExpection if no `OPENCV_TEST_DATA_PATH` is set
2) if it set then `fileData` MUST follows up to correct file. If path is incorrect that it MUST fail test.

The question is: do we need to check `initTestDataPathSilent` before fileData failed - or in another words: is my PR (by link above) behavior correct?

Comment (alalek) [reply modules/gapi/test/cpu/gapi_ocv_stateful_kernel_tests.cpp:183]: > throw SkipExpection if no OPENCV_TEST_DATA_PATH is set

`OPENCV_TEST_DATA_PATH` points to required test data (opencv_extra/testdata).
Test which requires data from opencv_extra must fail in that case.
Passed test (even through skip) is wrong behavior.

Test may be skipped if **optional** test data is missing, e.g. DNN models (it is done because full dataset is large).

---

G-API tests SHOULD NOT use `OPENCV_TEST_DATA_PATH` directly.
G-API tests SHOULD use `OPENCV_DNN_TEST_DATA_PATH` once during initialization (see "dnn" module).

**Comment by mpashchenkov** [comment]
@alalek, removed `initTestDataPath()` logic from gapi module. `ts module` set `OPENCV_TEST_DATA_PATH` variable for all tests.

**Review by sivanov-work — APPROVED** [review]
LGTM, if UTs behave as expected
It looks like other subsequent pending PRs (oneVPL) with new unit tests required to be rebased after this PR merged

**Comment by mpashchenkov** [comment]
@alalek, can it be merged? Is it the expected test behavior for you?

**Review by alalek — APPROVED** [review]
LGTM :+1:

**Pull request merged**
